lib/Transforms/IPO/StructFieldCacheAnalysisImpl.h

```diff
@@ -30,20 +31,33 @@
 #include "llvm/Transforms/IPO/StructFieldCacheAnalysis.h"
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 using namespace llvm;
 
+#define DEBUG_TYPE "struct-analysis"
+#define DEBUG_TYPE_IR "struct-analysis-IR"
+#define DEBUG_TYPE_STATS "struct-analysis-detailed-stats"
+#define DEBUG_TYPE_FRG "struct-analysis-FRG"
+#define DEBUG_TYPE_CPG "struct-analysis-CPG"
+
 namespace llvm {
 typedef unsigned FieldNumType;
 typedef std::pair<const StructType *, FieldNumType> StructInfoMapPairType;
-typedef uint64_t ExecutionCountType;
+typedef uint64_t ProfileCountType;
 typedef std::vector<FieldNumType> FieldNumArrayType;
-
+typedef double ExecutionCountType;
+typedef double DataBytesType;
```

Comment:
Because the (count, distance) relationship between two fields can be a weighted average of multiple paths. double is used to hold this result for better precision. But I know it's not good for compiler to do floating calculation. We should use integers later but not in this version.